src/Interpreters/InterpreterCreateQuery.cpp

```diff
@@ -1259,8 +1268,22 @@ bool InterpreterCreateQuery::doCreateTable(ASTCreateQuery & create,
         if (create.if_not_exists && getContext()->tryResolveStorageID({"", create.getTable()}, Context::ResolveExternal))
             return false;
 
+        create.setDatabase(DatabaseCatalog::TEMPORARY_DATABASE);
```

Comment:
Temporary tables do not belong to any database from a user's point of view, so we probably should not set it in the create query

Reply:
Good point. Corrected it.